corehq/apps/linked_domain/updates.py

```diff
@@ -95,3 +100,21 @@ def update_user_roles(domain_link):
 
         role_json.update(role_def)
         UserRole.wrap(role_json).save()
+
+
+def _convert_web_apps_permissions(domain_link, master_results):
+    linked_apps_by_master = {
+        app.master: app._id
+        for app in get_docs_in_domain_by_class(domain_link.linked_domain, LinkedApplication)
+    }
```

Comment:
Is it possible for two linked apps to have the same master? If so, I think app._id should be in a list, and you can use `extend()` instead of `append()` below.

Reply:
Hmm, good question. I guess it's possible (by who would want to do that????). I'll update.

Reply:
Actually I'm not going to change it. I think that's a rare situation that's only likely in testing.